projections/weather/lib/weather/schema.rb

```diff
@@ -20,77 +20,66 @@ module Schema
-
-    ListCitiesInput.add_member(:next_token, 'nextToken', Prelude::String)
-    ListCitiesInput.add_member(:page_size, 'pageSize', Prelude::Integer)
+    ListCitiesInput.add_member(:next_token, ShapeRef.new(target: Prelude::String, location: 'nextToken'))
+    ListCitiesInput.add_member(:page_size, ShapeRef.new(target: Prelude::Integer, location: 'pageSize'))
     ListCitiesInput.type = Types::ListCitiesInput
-
-    ListCitiesOutput.add_member(:next_token, 'nextToken', Prelude::String)
-    ListCitiesOutput.add_member(:items, 'items', CitySummaries, traits: {"smithy.api#required" => {}})
+    ListCitiesOutput.add_member(:next_token, ShapeRef.new(target: Prelude::String, location: 'nextToken'))
+    ListCitiesOutput.add_member(:items, ShapeRef.new(target: CitySummaries, location: 'items', traits: {"smithy.api#required" => {}}))
     ListCitiesOutput.type = Types::ListCitiesOutput
-
-    NoSuchResource.add_member(:resource_type, 'resourceType', Prelude::String, traits: {"smithy.api#required" => {}})
+    NoSuchResource.add_member(:resource_type, ShapeRef.new(target: Prelude::String, location: 'resourceType', traits: {"smithy.api#required" => {}}))
     NoSuchResource.type = Types::NoSuchResource
 
     SERVICE = ServiceShape.new do |service|
       service.id = "example.weather#Weather"
       service.name = "Weather"
       service.version = "2006-03-01"
-      service.traits = {"smithy.api#paginated" => {"inputToken" => "nextToken", "outputToken" => "nextToken", "pageSize" => "pageSize"}}
+      service.traits = {}
       service.add_operation(:get_city, OperationShape.new do |operation|
         operation.id = "example.weather#GetCity"
         operation.name = "GetCity"
-        operation.input = GetCityInput
-        operation.output = GetCityOutput
+        operation.input = ShapeRef.new(target: GetCityInput)
+        operation.output = ShapeRef.new(target: GetCityOutput)
+        operation.errors << ShapeRef.new(target: NoSuchResource)
         operation.traits = {"smithy.api#readonly" => {}}
-        operation.errors << NoSuchResource
       end)
       service.add_operation(:get_current_time, OperationShape.new do |operation|
         operation.id = "example.weather#GetCurrentTime"
         operation.name = "GetCurrentTime"
-        operation.input = Prelude::Unit
-        operation.output = GetCurrentTimeOutput
+        operation.input = ShapeRef.new(target: Prelude::Unit)
+        operation.output = ShapeRef.new(target: GetCurrentTimeOutput)
         operation.traits = {"smithy.api#readonly" => {}}
       end)
       service.add_operation(:get_forecast, OperationShape.new do |operation|
         operation.id = "example.weather#GetForecast"
         operation.name = "GetForecast"
-        operation.input = GetForecastInput
-        operation.output = GetForecastOutput
+        operation.input = ShapeRef.new(target: GetForecastInput)
+        operation.output = ShapeRef.new(target: GetForecastOutput)
```

Comment:
What benefits does `ShapeRef` bring when you wrap it with Input and Output shapes? The only that is obvious to me is the ability to add/access the `metadata` attribute.